### PR TITLE
cmd/geth: update copydb to use default freezer if not specified

### PR DESCRIPTION
**Pull request by david-drinn**
The copydb command requires a second argument, but the help doesn't reflect that. This attempts to resolve that by updating the help, but also using the default freezer directory if not specified.

This is an alt to https://github.com/ethereum/go-ethereum/pull/22203, which left function alone and just updated the help. They're mutually exclusive PRs.

### CODE REVIEW

File: cmd/geth/chaincmd.go [hunk 1]
```diff
@@ -448,9 +450,13 @@ func copyDb(ctx *cli.Context) error {
 	if len(ctx.Args()) < 1 {
 		utils.Fatalf("Source chaindata directory path argument missing")
 	}
-	if len(ctx.Args()) < 2 {
-		utils.Fatalf("Source ancient chain directory path argument missing")
+
+	// Get user-defined ancient directory if available, otherwise use default.
+	freezer := ""
```
Comment (rjl493456442) [inline cmd/geth/chaincmd.go:455]: It's wrong. The default freezer path is `freezer = filepath.Join(root, "ancient")`. Check https://github.com/ethereum/go-ethereum/blob/master/node/node.go#L578 for more details

**Review by rjl493456442 — CHANGES_REQUESTED** [review]
This version is better I think. But the code is not correct. Please fix it.

File: cmd/geth/chaincmd.go (hunk 1 above)
Comment (david-drinn) [reply cmd/geth/chaincmd.go:455]: I think this is resolved.

**Comment by david-drinn** [comment]
I think I see my mistake. Will fix it.

**Comment by david-drinn** [comment]
> I think I see my mistake. Will fix it.

Hmm then again, maybe I don't see it. I'm pretty new to Golang and thought it was a type-mismatch issue, but I don't think so.

**Comment by david-drinn** [comment]
Ready for review.

**Comment by karalabe** [comment]
We've debated this and we've figured there's no reason to fix copydb, rather we should just drop it. The reasoning is that copydb was a hacky attempt at state pruning (copy into a new folder, delete old folder, move back). @rjl493456442 has a new PR (https://github.com/ethereum/go-ethereum/pull/21724) that solves database pruning much more elegantly, which we'll ship in the next release, so there's no point to keep hacking the old hack.

**Comment by karalabe** [comment]
@david-drinn Would you be interested in deleting all notions of copydb in a new PR?

**Comment by david-drinn** [comment]
@karalabe yeah for sure. Where should I branch from, just current master for now, and then merge later? Or is there a cleaner branch to start from?

**Comment by holiman** [comment]
If nothing else prevents it, it's always easier to start off from master, then there's less chance of needing to rebase + fixup  later due to conflicts.

**Pull request closed without merging**